src/args.rs

```diff
@@ -0,0 +1,47 @@
+#[derive(Debug, PartialEq, Eq)]
+pub struct Args {
+    pub command: String
+}
+
+impl Args {
+    pub fn parse(raw_args: Vec<String>) -> Result<Args, String> {
+        match raw_args.len() {
+            0 => Err(String::from("Please pass remote command.")), // TODO more user friendly message, for now it's consistent with Bash version.
+            _ => Ok(Args {
+                command: {
+                    let str: String = raw_args
+                        .iter()
+                        .cloned()
```

Comment:
Because you own `raw_args` you can avoid cloning by calling `.into_iter()`. This will consume `raw_args` and turn it into an iterator that yields owned Strings.